src/Manipulators/Size.php

```diff
@@ -271,102 +270,93 @@ public function runResize(Image $image, $fit, $width, $height)
      *
-     * @return Image The manipulated image.
+     * @return ImageInterface The manipulated image.
      */
-    public function runContainResize(Image $image, $width, $height)
+    public function runContainResize(ImageInterface $image, $width, $height): ImageInterface
     {
-        return $image->resize($width, $height, function ($constraint) {
-            $constraint->aspectRatio();
-        });
+        return $image->scale($width, $height);
     }
 
     /**
      * Perform max resize image manipulation.
      *
-     * @param Image $image  The source image.
-     * @param int   $width  The width.
-     * @param int   $height The height.
+     * @param ImageInterface $image  The source image.
+     * @param int            $width  The width.
+     * @param int            $height The height.
      *
-     * @return Image The manipulated image.
+     * @return ImageInterface The manipulated image.
      */
-    public function runMaxResize(Image $image, $width, $height)
+    public function runMaxResize(ImageInterface $image, $width, $height): ImageInterface
     {
-        return $image->resize($width, $height, function ($constraint) {
-            $constraint->aspectRatio();
-            $constraint->upsize();
-        });
+        return $image->resize($width, $height);
     }
 
     /**
      * Perform fill resize image manipulation.
      *
-     * @param Image $image  The source image.
-     * @param int   $width  The width.
-     * @param int   $height The height.
+     * @param ImageInterface $image  The source image.
+     * @param int            $width  The width.
+     * @param int            $height The height.
      *
-     * @return Image The manipulated image.
+     * @return ImageInterface The manipulated image.
      */
-    public function runFillResize($image, $width, $height)
+    public function runFillResize(ImageInterface $image, $width, $height): ImageInterface
     {
         $image = $this->runMaxResize($image, $width, $height);
 
-        return $image->resizeCanvas($width, $height, 'center');
+        return $image->resizeCanvas($width, $height, 'ffffff', 'center');
     }
 
     /**
      * Perform fill-max resize image manipulation.
      *
-     * @param Image $image  The source image.
-     * @param int   $width  The width.
-     * @param int   $height The height.
+     * @param ImageInterface $image  The source image.
+     * @param int            $width  The width.
+     * @param int            $height The height.
      *
-     * @return Image The manipulated image.
+     * @return ImageInterface The manipulated image.
      */
-    public function runFillMaxResize(Image $image, $width, $height)
+    public function runFillMaxResize(ImageInterface $image, $width, $height): ImageInterface
     {
-        $image = $image->resize($width, $height, function ($constraint) {
-            $constraint->aspectRatio();
-        });
+        $image = $image->resize($width, $height);
```

Comment:
```suggestion
        $image = $image->scale($width, $height);
```

Reply:
Hmm, I can not find any differences using query `?h=400&w=400&size=fill-max`.

![grafik](https://github.com/thephpleague/glide/assets/2162994/38a7a5f5-5a23-4f4f-89d4-04596eead763)

Reply:
Are these images 400x400?